Source/Evolve/WarpXEvolve.cpp

```diff
@@ -526,6 +418,124 @@ WarpX::OneStep_nosub (Real cur_time)
     ExecutePythonCallback("afterEsolve");
 }
 
+bool WarpX::checkStopSimulation (amrex::Real cur_time)
+{
+    m_exit_loop_due_to_interrupt_signal = SignalHandling::TestAndResetActionRequestFlag(SignalHandling::SIGNAL_REQUESTS_BREAK);
+    if (cur_time >= stop_time - 1.e-3*dt[0] || m_exit_loop_due_to_interrupt_signal) {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
```

Comment:
```suggestion
    return (cur_time >= stop_time - 1.e-3*dt[0])  ||
        m_exit_loop_due_to_interrupt_signal;
```
`clang-tidy` suggests doing this. l think that's a good idea: it makes this section less verbose.